app/controllers/carts_controller.rb

```diff
@@ -16,9 +16,29 @@ def add_product
     end
   end
 
+  def checkout
+    create_new_order unless unfinished_checkout?
+  end
+
   private
 
   def cart
     @cart ||= current_user.cart
   end
+
+  def unfinished_checkout?
+    if current_user.orders.find_by(state: :new).present?
+      redirect_to new_check_out_path, alert: 'You must finish check out'
+    end
+  end
+
+  def create_new_order
+    order = current_user.orders.build
+    response = CartServices::CheckOut.new.call cart, order
```

Comment:
Myślałem o przeniesieniu `orders.build` do CheckOut serwisu ale wtedy CheckOut wiedziałby chyba za dużo o current_user.
```suggestion
    response = CartServices::CheckOut.new.call cart, current_user
```